components/setup/DockerInstructions.tsx

```diff
@@ -0,0 +1,207 @@
+"use client";
+
+import axios from 'axios';
+import type { Session } from "next-auth";
+import React, { useEffect, useState } from 'react';
+
+import { RepoIdentifier } from '../../types/repository';
+import { getUserRepositories } from '../../utils/providerAPI/getUserRepositories';
+import Button from '../Button';
+import CodeWithCopyButton from './CodeWithCopyButton';
+import InstructionsToGeneratePersonalAccessToken from './InstructionsToGeneratePersonalAccessToken';
+
+interface DockerInstructionsProps {
+	userId: string;
+	selectedProvider: string;
+	selectedInstallationType: string;
+	session: Session | null;
+}
+
+interface SetupReposApiBodyArgs {
+	owner: string,
+	provider: string,
+	repos: string [],
+	installationId: string
+}
+
+async function getUserReposForProvider(session: Session, targetProvider: string) {
+	const allRepos = await getUserRepositories(session);
+	// Filter repos based on the targetProvider
+	const filteredRepos = allRepos.filter(repo => repo.repo_provider === targetProvider);
+	return filteredRepos;
+}
+
+function formatRepoListInSaveSetupArgsForm(repos: RepoIdentifier[], install_id: string) {
+	// Group by repo_owner and generate setup args
+	const setupArgsMap: Map<string, SetupReposApiBodyArgs> = new Map();
+	repos.forEach(repo => {
+		const key = repo.repo_owner;
+		if (!setupArgsMap.has(key)) {
+			setupArgsMap.set(key, {
+				owner: repo.repo_owner,
+				provider: repo.repo_provider,
+				repos: [],
+				installationId: install_id
+			});
+		}
+		setupArgsMap.get(key)!.repos.push(repo.repo_name);
```

Comment:
```suggestion
		setupArgsMap.get(key)?.repos.push(repo.repo_name);
```

Shouldn't this be handled with safety?